src/nat/runtime/session.py

```
@@ -37,67 +41,374 @@
+                await asyncio.wait_for(self._shutdown_event.wait(),
+                                       timeout=self._per_user_session_cleanup_interval.total_seconds())
+                # If we get here, shutdown was signaled
+                break
+            except TimeoutError:
+                # Timeout means it's time to run cleanup
+                try:
+                    cleaned = await self._cleanup_inactive_per_user_builders()
+                    if cleaned > 0:
+                        logger.info(f"Cleaned up {cleaned} inactive per-user builder(s)")
+                except Exception as e:
+                    logger.exception(f"Error during periodic cleanup: {e}")
```

Ruff (0.14.8)

Redundant exception object included in `logging.exception` call

(TRY401)